src/RegisterHandler.php

```diff
@@ -0,0 +1,143 @@
+<?php declare(strict_types=1);
+
+namespace SilverStripe\TOTP;
+
+use OTPHP\TOTP;
+use OTPHP\TOTPInterface;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Environment;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\MFA\Exception\AuthenticationFailedException;
+use SilverStripe\MFA\Method\Handler\RegisterHandlerInterface;
+use SilverStripe\MFA\Service\EncryptionAdapterInterface;
+use SilverStripe\MFA\Store\StoreInterface;
+
+/**
+ * Handles registration requests using a time-based one-time password (TOTP) with the silverstripe/mfa module.
+ */
+class RegisterHandler implements RegisterHandlerInterface
+{
+    use Configurable;
+
+    /**
+     * The link to SilverStripe user help documentation for this authenticator.
+     * @todo add it
+     *
+     * @config
+     * @var string
+     */
+    private static $user_help_link = '';
+
+    public function start(StoreInterface $store): array
+    {
+        $totp = $this->getTotp($store);
+
+        $totp->setLabel($store->getMember()->Email);
+        $totp->setIssuer('SilverStripe');
```

Comment:
I feel like this is probably better suited as the site name? Does it show in the authenticator after registering?

Reply:
Yeah, you're right. This was a placeholder I guess

Reply:
I've updated this to use the SiteConfig title in the next PR - is that OK?